### PR TITLE
nucleo_h745zi_q remove pinmux dependency

### LINKED ISSUE #29675: Remove pinmux dependency on STM32 boards
Removing pinmux.c of each of ST's Zephyr-Tree boards can be done now after DT approach were introduced. My new PR #29674 shows first feasible sample.

### PR DESCRIPTION
**Pull request by KozhinovAlexander**
Remove pinmux dependency of the nucleo_h745zi_q board, since it is no more needed after move to DT approach.

### CODE REVIEW

**Review by gmarull — CHANGES_REQUESTED** [review]
pinmux files should not be removed until they are officially deprecated. I think it's better to wait for that moment and delete all of them at once.

**Comment by Nukersson** [comment]
> pinmux files should not be removed until they are officially deprecated. I think it's better to wait for that moment and delete all of them at once.

Do you mean to wait until pinmux can be removed on all boards simultaneously? Or you mean leaving empty `pinmux.c` within this PR should be okay?

File: boards/arm/nucleo_h745zi_q/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,5 @@
 # SPDX-License-Identifier: Apache-2.0
 
 if(CONFIG_PINMUX)
-zephyr_library()
-zephyr_library_sources(pinmux.c)
 zephyr_library_include_directories(${ZEPHYR_BASE}/drivers)
 endif()
```
Comment (ABOSTM) [inline boards/arm/nucleo_h745zi_q/CMakeLists.txt:5]: If aim is to remove pinmux , can't we get ride of 
```
if(CONFIG_PINMUX) 
```
?

Comment (Nukersson) [reply boards/arm/nucleo_h745zi_q/CMakeLists.txt]: I can confirm, that removing of above `if` in `CMakeLists.txt` works also fine and is possible.

**Comment by Nukersson** [comment]
Closed due to @erwango notice in #29675. I will keep this branch anyway for a while.

**Pull request closed without merging**